api/src/test/java/org/openmrs/module/patientqueueing/api/PatientQueueingServiceTest.java

```diff
@@ -22,140 +22,186 @@
-		patientQueue.setQueueNumber("QN-001");
-		patientQueue.setEncounter(Context.getEncounterService().getEncounter(10000));
-		patientQueue.setLocationFrom(location);
-		patientQueue.setLocationTo(location);
-		patientQueue.setPriority(0);
-		patientQueue.setPriorityComment("Emergency");
-		patientQueueingService.savePatientQue(patientQueue);
-		
-		List<PatientQueue> newPatientQueueList = patientQueueingService.getPatientQueueList(null, null, null, null, null,
-		    null, null);
-		
-		Assert.assertEquals(originalPatientQueueList.size() + 1, newPatientQueueList.size());
-		
-		Assert.assertEquals("QN-001", newPatientQueueList.get(0).getQueueNumber());
-		
-	}
-	
-	@Test
-	public void savePatientQueue_shouldCompleteExistingPatientQueueWithSamePatientAndLocation() throws Exception {
-		
-		PatientQueueingService patientQueueingService = Context.getService(PatientQueueingService.class);
-		
-		Patient patient = Context.getPatientService().getPatient(10000);
-		
-		Location location = Context.getLocationService().getLocation(1);
-		
-		List<PatientQueue> originalPatientQueueList = patientQueueingService.getPatientQueueList(null, null, null, null,
-		    null, null, null);
-		
-		PatientQueue patientQueue = new PatientQueue();
-		patientQueue.setPatient(patient);
-		patientQueue.setStatus(PatientQueue.Status.PENDING);
-		patientQueue.setQueueNumber("QN-001");
-		patientQueue.setEncounter(Context.getEncounterService().getEncounter(10000));
-		patientQueue.setLocationFrom(location);
-		patientQueue.setLocationTo(location);
-		patientQueue.setPriority(0);
-		patientQueue.setPriorityComment("Emergency");
-		PatientQueue patientQueue1 = patientQueueingService.savePatientQue(patientQueue);
-		
-		PatientQueue patientQueue2 = new PatientQueue();
-		patientQueue2.setPatient(patient);
-		patientQueue2.setStatus(PatientQueue.Status.PENDING);
-		patientQueue2.setQueueNumber("QN-002");
-		patientQueue2.setEncounter(Context.getEncounterService().getEncounter(10000));
-		patientQueue2.setLocationFrom(location);
-		patientQueue2.setLocationTo(location);
-		patientQueue2.setPriority(0);
-		patientQueue2.setPriorityComment("Emergency");
-		patientQueueingService.savePatientQue(patientQueue2);
-		
-		Assert.assertEquals(PatientQueue.Status.COMPLETED, patientQueue1.getStatus());
-		
-	}
+
+    private static final String QUEUE_STANDARD_DATASET_XML = "org/openmrs/module/patientqueueing/standardTestDataset.xml";
+
+    private static Logger logger = LoggerFactory.getLogger(PatientQueueingServiceTest.class);
+
+    @Before
+    public void initialize() throws Exception {
+        executeDataSet(QUEUE_STANDARD_DATASET_XML);
+    }
+
+    @InjectMocks
+    PatientQueueingServiceImpl patientQueueingService;
+
+    @Mock
+    PatientQueueingDao dao;
+
+    @Before
+    public void setupMocks() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void getIncompletePatientQueue_shouldReturnInCompletePatientQueue() throws Exception {
```

Comment:
shouldSaveAndReturn....

Reply:
I dont seem to understand where you put your comment